5.3.js

```diff
@@ -0,0 +1,28 @@
+
+function average(array) {
+  function plus(a, b) { return a + b; }
+  return array.reduce(plus) / array.length;
+}
+
+function groupBy(array, groupOf) {
+  var groups = {};
+  array.forEach(function(element) {
+    var groupName = groupOf(element);
+    if (groupName in groups)

```

Comment:
dodatkowo: 
generalnie w ten sposob tworzysz dwa postepowania logiczne, w zaleznosci od tego jaki typ danych dostaniesz.

jest to dosc slabe rozwiazanie, bo jesli typow danych bedzie wiecej albo jezeli prace na nich beda czasochlonne to kod rozrasta sie do niebotycznych rozmiarow a my mamy problem z operowaniem na nim.

dlatego sugestia jest najpierw standaryzowac dane wejsciowe. w tym przypadku:

```javascript
if (!Array.isArray(groups[groupName])) {
  groups[groupName] = [];
}
// i juz caly czas operujesz na arrayach
```